clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/cache/CacheRepository.java

```diff
@@ -169,6 +169,36 @@ private CloudFoundryLoadBalancer loadBalancerFromCacheData(CacheData lbData, Det
         findServerGroupsByKeys(lbData.getRelationships().get(SERVER_GROUPS.getNs()), Detail.NONE));
   }
 
+  public Set<CloudFoundryLoadBalancer> findLoadBalancersByClusterKeys(
+      Collection<String> keys, Detail detail) {
+    Set<String> serverGroupKeys =
```

Comment:
Could we go directly to `serverGroups` without using `clusters` by using `cacheView.filterIdentifiers(SERVER_GROUPS.getNs(), Keys.getClusterKey("*", app, "*"))`?